cmd/ci-operator/main.go

```diff
@@ -411,6 +414,31 @@ func (o *options) Complete() error {
 		log.Printf("Resolved job spec:\n%s", string(job))
 	}
 
+	// resolve releases we're asked to resolve and propagate their values
+	for name, release := range o.configSpec.Releases {
+		var value string
+		var err error
+		switch {
+		case release.Candidate != nil:
+			value, err = candidate.ResolvePullSpec(*release.Candidate)
+		case release.Release != nil:
+			value, err = official.ResolvePullSpec(*release.Release)
+		case release.Prerelease != nil:
+			value, err = prerelease.ResolvePullSpec(*release.Prerelease)
+		}
+		if err != nil {
+			return results.ForReason("resolving_release").ForError(fmt.Errorf("failed to resolve release %s: %v", name, err))
+		}
+		log.Printf("Resolved release %s to %s", name, value)
+		env := fmt.Sprintf("RELEASE_IMAGE_%s", strings.ToUpper(name))
+		if current := os.Getenv(env); current != "" {
+			return results.ForReason("exposing_release").ForError(fmt.Errorf("could not resolve release and set to %s, as that environment was already provided", env))
+		}
+		if err := os.Setenv(env, value); err != nil {
+			return results.ForReason("exposing_release").ForError(fmt.Errorf("failed to expose release %s as %s: %v", name, env, err))
+		}
```

Comment:
This is OK for a first implementation, but it would be great to pass the values explicitly instead of indirectly via the environment (and all the background knowledge required to understand how that works). I think the environment override was added specifically for releases, so maybe we'll be able to remove it altogether.

A comment here explaining (or at least giving a hint) that this will eventually propagate to tests would be helpful.

Reply:
Yeah, agreed. I was hoping to mess with how we pass these around in the explicit dependencies card as someone may be able to ask for e.g. `release:foo` as a dependency of their step.